lib/OpenQA/Script/CloneJob.pm

```diff
@@ -206,38 +206,50 @@ sub openqa_baseurl {
 }
 
 sub clone_job {
-    my ($jobid, $options, $clone_map, $depth) = @_;
-    $clone_map //= {};
-    $depth     //= 0;
+    my ($jobid, $options, $clone_map, $depth, $parent_jobid) = @_;
+    $clone_map    //= {};
+    $depth        //= 0;
+    $parent_jobid //= 0;
     return $clone_map->{$jobid} if defined $clone_map->{$jobid};
 
     my ($ua, $local, $local_url, $remote, $remote_url) = create_url_handler($options);
-    my $job = clone_job_get_job($jobid, $remote, $remote_url, $options);
-    if ($job->{parents}) {
-        my ($chained, $directly_chained, $parallel);
-        unless ($options->{'skip-deps'}) {
-            unless ($options->{'skip-chained-deps'}) {
-                $chained          = $job->{parents}->{Chained};
-                $directly_chained = $job->{parents}->{'Directly chained'};
+    my $job       = clone_job_get_job($jobid, $remote, $remote_url, $options);
+    my @job_types = 'parents';
+    push @job_types, 'children' if $options->{'clone-children'} && $depth == 0;
+    my $child_list;
+    for my $job_type (@job_types) {
+        if ($job->{$job_type}) {
+            my ($chained, $directly_chained, $parallel);
+            unless ($options->{'skip-deps'}) {
+                unless ($options->{'skip-chained-deps'}) {
+                    $chained          = $job->{$job_type}->{Chained};
+                    $directly_chained = $job->{$job_type}->{'Directly chained'};
+                }
+                $parallel = $job->{$job_type}->{Parallel};
+            }
```

Comment:
This block becomes a bit too convoluted. I suggest you extract that into a function which you can also get more easily covered in tests. This way you can reduce the indendation level as well.

Reply:
Do you want a function for the lines between L223 to L229 or for L222 to L252?

Reply:
I leave it up to you which way you think is more easy to understand and looks prettier :) Obviously, level of indentation is just a superficial symptom of how easy code is to understand or maintain. So functions should have a proper name so that each level of complexity can be read and understood on its own which is easier than trying to understand all levels of complexity/indention here.

Reply:
I created a function `get_deps` to clear the code a little bit.